src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Temperature GetTemperature()
+        {
+            Measure();
+            return new Temperature(_temperature, UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+        }
+
+        /// <summary>
+        /// Gets the current humidity reading from the sensor.
+        /// </summary>
+        /// <returns>Temperature reading</returns>
+        public Ratio GetHumidity()
+        {
+            Measure();
+            return new Ratio(_humidity, UnitsNet.Units.RatioUnit.Percent);
+        }
+
+        /// <summary>
+        /// Perform sequence to retrieve current readings from device
+        /// </summary>
+        private void Measure()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                (byte)Command.Measure,
+                0x33, // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+
+            while (IsBusy())
+            {
+                Thread.Sleep(10);
+            }
+
+            buffer = stackalloc byte[6];
+            _i2cDevice.Read(buffer);
+
+            Int32 rawHumidity = (buffer[1] << 12) | (buffer[2] << 4) | (buffer[3] >> 4);
+            _humidity = (rawHumidity * 100.0) / 0x100000;
+            Int32 rawTemperature = ((buffer[3] & 0xF) << 16) | (buffer[4] << 8) | buffer[5];
+            _temperature = ((rawTemperature * 200.0) / 0x100000) - 50;
+        }
+
+        /// <summary>
+        /// Perform soft reset command sequence
+        /// </summary>
+        private void SoftReset()
+        {
+            _i2cDevice.WriteByte((byte)Command.SoftRest);
+            Thread.Sleep(20); // reset requires 20ms at most, c.f. datasheet version 1.1, ch. 5.5
+        }
+
+        /// <summary>
+        /// Perform calibration command sequence
+        /// </summary>
+        private void Calibrate()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                (byte)Command.Calibrate,
+                0x08, // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+            Thread.Sleep(10); // wait 10ms c.f. datasheet, version 1.1, ch. 5.4
+        }
+
+        private byte GetStatusByte()
+        {
+            _i2cDevice.WriteByte(0x71);
+            Thread.Sleep(10); // whithout this delay the reading the status fails often.
+            byte status = _i2cDevice.ReadByte();
+            return status;
+        }
+
+        private bool IsBusy()
+        {
+            return (GetStatusByte() & (byte)StatusBit.Busy) == (byte)StatusBit.Busy;
+        }
+
+        private bool IsCalibrated()
+        {
+            return (GetStatusByte() & (byte)StatusBit.Calibrated) == (byte)StatusBit.Calibrated;
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
```

Comment:
Since this class is not sealed, it should use the correct dispose pattern (with an overloadable `Dispose(bool disposing)` member)

Reply:
Ok. I'll change this.